### ISSUE TITLE
Modify document page, document feed and commissioner page wagtail templates

### ISSUE THREAD

**Issue by JonellaCulmer** [fecgov/fec-cms#3869]
**What we're after:** To make it easier for users to find documents and other content on commissioner pages, we need to make adjustments that support the addition of document feeds. 

Changes:
Commissioner page - Ensure we can include a button to link to a document feed page

Document feed page - Needs to sort by three things: category, subject, and year, external links, PDFs, video all need to be options for items in the feed and display with the correct icons. 

Document page - Change template to add new field for Page URL, to accommodate external links, format external links to appear as such on the document feed page (may be a document feed template change instead), dropdown for a type of link (PDF, wagtail page, external link in the template

### Designs
![Screen Shot 2020-06-26 at 3.31.25 PM.png](https://images.zenhubusercontent.com/59b1a5f3b0222d5de476d238/a7a2bdf4-9ef2-4870-83dd-89ed02ddff54)

![Screen Shot 2020-06-26 at 3.31.34 PM.png](https://images.zenhubusercontent.com/59b1a5f3b0222d5de476d238/8f441674-ea15-4afb-9dc0-1bf0b53c2454)

**Comment by rfultz** [comment]
(studying site structure, tinkering, breaking things (locally))

**Comment by rfultz** [comment]
Been working through options, breaking things. Ran through options and scope with John and Jonella on Friday. Now to implement it!

**Comment by rfultz** [comment]
Have created the CommissionerItem but also working my way through a StreamField for the DocumentChooserBlock and ParentalKey functionality for linking commissioners to their items (statements, opinions, etc)

**Comment by JonellaCulmer** [comment]
@rfultz When we get to the testing for this, we need to make sure to include content since they live in Wagtail and we use these templates elsewhere. 

cc: @dorothyyeager @AmyKort

**Comment by rfultz** [comment]
Big discussion yesterday about taxonomy functionality. Started the overhaul

**Comment by rfultz** [comment]
We can start reviewing code now, but there will likely be changes after #3947

**Comment by rfultz** [comment]
After yesterday's discussion, we're looking at a temporary solution to demo next week. downsides: would work only with this one page and would require extensive code editing by site authors

**Comment by rfultz** [comment]
@JonellaCulmer here's a grab of the what I've got the results text added. Meh. I put little thought into it, knowing you were coming along behind me :) 
![image](https://user-images.githubusercontent.com/26720877/93376727-06854900-f828-11ea-928c-b34973755577.png)

I wonder if we should update this ticket with the new approach rather than the first attempt

**Comment by JonellaCulmer** [comment]
Thanks, @rfultz! 

And that sounds good to me too, or rather opening a new fresh ticket with just the solution we're working on and close this one in favor of that one.

**Comment by JonellaCulmer** [comment]
@rfultz How much would you hate me if we introduced another line of text that says Results when the fields are used? 

I’d also like to change the color of the highlight to match other sections on the site (example: https://www.fec.gov/help-candidates-and-committees/filing-reports/redesignating-and-reattributing-contributions/) I’m asking James Jones exactly what color that is, because it’s not officially in our pattern library. I’ve estimated it in my mockup with a 50% opacity of our red.

A couple other things, we need to add a dividing line above the filters to make it more clear that there’s a separation in the content that will be searched. Nothing above the line/search box will be searched. 

And Tom talked about the grouping of the filters, so I rearranged them in the mockup. 

Let me know what you think about the results portion. 
<img width="604" alt="Screen Shot 2020-09-16 at 4 48 05 PM" src="https://user-images.githubusercontent.com/31663028/93391508-40147f00-f83d-11ea-9ccf-1a8e21f8562e.png">

**Comment by rfultz** [comment]
@JonellaCulmer would the Results headline appear only when a filter has been used? And then disappear again when all of the filters are back to 'All' and the text search is blank?

@dorothyyeager, I'm not sure how the rest of the site's searches work, but the text search field is only looking for content on this page and only lower on the page than these fields (the Commissioner Items blocks). There is no form submission, no page reload, no way to link to a filtered search of this page—it's purely done in-browser

**Comment by rfultz** [comment]
Until we get the color code, I just grabbed the color from the Feedback tab :)

![image](https://user-images.githubusercontent.com/26720877/93605543-e45d0980-f994-11ea-9ffb-66807f9a9a93.png)

**Comment by rfultz** [comment]
Is there a difference between these two?

The left-aligned headline with a pipe separating it from a kind of subhead / label, are we using that anywhere else?

**Comment by JonellaCulmer** [comment]
@rfultz No, we're not using that elsewhere. It's a slight modification on the first one I sent, in an attempt to make it a bit easier to spot the results. It's different from the guidance search where we have 100% page width to work with. The tighter coupling may improve the usability of the results text. 

The difference between the two is there is a line below results in the first and no line in the second. The line seems heavy to me in this space and I wanted to get your thoughts on that.

**Comment by dorothyyeager** [comment]
@JonellaCulmer @rfultz You guys are doing a great job thinking this through.

The highlight field is from "reporting examples" in case you need to find a template with it in Wagtail. Let me know if I should make a ticket to put it in our pattern library.

I like the new order of the fields.

For the example text - I think we need to show an example of a two-word text search; something like "dark money" or "foreign national" would work. That way users will see whether they need quotes or AND or whatever they need to get the phrase.

**Comment by JonellaCulmer** [comment]
@dorothyyeager Okay. Combined with my desire to show that partial words will work, what do you think about the following:

Examples: dark money, dark, dar

cc: @rfultz

**Comment by rfultz** [comment]
For the text search, I'd be wary of suggesting more complicated searches. The logic is that the characters are in that order but could get complicated if we cross otherwise-invisible tags (`<b>`, `</i>`, `<em>`, `</a>`, etc). I've set the field not to allow punctuation to limit breaking things, like if they search for code that could execute something. This is more of a "find in page" than search as most of us think of search.

Just a reminder: the results will update as the user types each character. As they type "Dark", it'll be filtered by "d", then "da", then "dar", in near-realtime.

**Comment by JonellaCulmer** [comment]
@rfultz @dorothyyeager Per our discussion. This is the example text we'll go with. 

<img width="749" alt="Screen Shot 2020-09-21 at 4 25 11 PM" src="https://user-images.githubusercontent.com/31663028/93817475-0b326e80-fc27-11ea-815e-d85a7a494ea1.png">

**Comment by rfultz** [comment]
Update: holding for details/content (holding for the content data, not the content team)

**Comment by jason-upchurch** [comment]
@rfultz  I moved this to innovation sprint in preparation for the next iteration. Please let me know if I need to reprioritize it--thank you!
